components/script/dom/htmllinkelement.rs

```diff
@@ -70,7 +70,7 @@ impl HTMLLinkElement {
         HTMLLinkElement {
             htmlelement: HTMLElement::new_inherited(local_name, prefix, document),
             rel_list: Default::default(),
-            parser_inserted: Cell::new(creator == ElementCreator::ParserCreated),
+            parser_inserted: Cell::new(creator == ElementCreator::ParserCreated(1)),
```

Comment:
Let's add a `is_parser_created` method to the ElementCreator type (`impl ParserCreated`) that performs a match. As written, this will cause incorrect results in cases where the element is not on line 1 of the HTML file it's created from.